### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by gelzinyte**
This should be pretty much complete up-to-date documentation.

### CODE REVIEW

**Comment by bernstei** [comment]
Generally looks really good, maybe a typo here and there.  Did you mention something problematic with some of the auto-generated module docs that I should look at?

**Comment by gelzinyte** [comment]
Thanks! 

Yes, docstring for functions in `wfl.generate.md`  aren't included - make complains about a circular reference. I can't figure out why - I can't spot the circular reference and the whole submodule is at the same level of depth as `wfl.calculators.orca` which has no problems. 

Note that the module files (`module.rst` and everything starting with `wfl.`) are auto-generated with, from `workflow/docs` folder:

```
sphinx-apidoc -o source ../wfl
```

I was going to come back to this later if the solution isn't immediately obvious to you.

**Comment by bernstei** [comment]
See https://github.com/sphinx-doc/sphinx/issues/10914

**Comment by bernstei** [comment]
I'm ready to merge this, and we can deal with `wfl.generate.md` separately, ideal via a sphinx bug fix.

**Comment by gelzinyte** [comment]
Ok. Let's merge this after PR #162

File: tests/calculators/test_orca.py [hunk 1]
```diff
@@ -88,8 +88,7 @@ def test_full_orca(tmp_path):
         atoms.get_potential_energy()
     except CalculationFailed:
         pass
-    # Temporarily: scratchdir isn't cleaned up, see issue 161. 
-    # assert not any(scratchdir.iterdir())
+    assert not any(scratchdir.iterdir())
```
[the remark this answers is not in the document]
Comment (gelzinyte) [reply tests/calculators/test_orca.py:91]: fixed, but in #162

File: wfl/calculators/wfl_fileio_calculator.py [hunk 2]
```diff
@@ -75,3 +76,8 @@ def clean_rundir(self, _default_keep_files, calculation_succeeded):
         if self._wfl_scratchdir is not None:
             for f in Path(self.directory).glob("*"):
                 shutil.move(f, self._cur_rundir)
+            if any(Path(self.directory).iterdir()):
```
[the remark this answers is not in the document]
Comment (gelzinyte) [reply wfl/calculators/wfl_fileio_calculator.py:79]: Fixed, but in #162

**Comment by bernstei** [comment]
I'm ready to merge, maybe after #162 is actually merged?

**Comment by bernstei** [comment]
I'll take a look at the conflicts.  Assuming they're simple, are we ready to merge this one as well?

**Comment by gelzinyte** [comment]
Thanks. Yes, do merge.

**Comment by bernstei** [comment]
> Thanks. Yes, do merge.

OK.  Once the tests pass, I'll do it.

**Pull request merged**